ExternalTest/rr_sim_tests/rr_sim_tests/utils/utils.py

```diff
@@ -78,47 +103,64 @@ def __init__(self, in_robot_namespace:str = '', publishing_num = PUBLISHING_NUM,
+            self._timer = self.create_timer(1.0 / publishing_freq, self.twist_robot)
 
             while rclpy.ok() and not self._is_pub_finished:
-                self._executor.spin_once(timeout_sec = 0)
-            print(f'Finished publishing Twist to {self.get_full_topic_name()}')
+                self._executor.spin_once(timeout_sec=0)
+            print(f"Finished publishing Twist to {self.get_full_topic_name()}")
 
     def get_full_topic_name(self):
-        return f'{self.get_namespace()}/{self._cmd_vel_topic}'
+        return f"{self.get_namespace()}/{self._cmd_vel_topic}"
 
     def __enter__(self):
         return self
 
     def __exit__(self, exep_type, exep_value, trace):
         if exep_type is not None:
-            raise Exception('Exception occured, value: ', exep_value)
-        self._executor.shutdown(timeout_sec = 0.5)
+            raise Exception("Exception occured, value: ", exep_value)
+        self._executor.shutdown(timeout_sec=0.5)
         self.destroy_node()
-        
+
     async def twist_robot(self):
-        if (self._twist_pub_count < self._publishing_num or self._publishing_num <= 0):
-            self.get_logger().info(f'Publishing to {self.get_full_topic_name()}: {self._twist}')
-            self._twist_pub.publish(self._twist)
+        if self._twist_pub_count < self._publishing_num or self._publishing_num <= 0:
+            self.get_logger().info(
+                f"Publishing to {self.get_full_topic_name()}: {self._twist}"
+            )
+            self.pub(self._twist)
             self._twist_pub_count += 1
         else:
             self._is_pub_finished = True
             self._timer.cancel()
         await asyncio.sleep(0)
-    
-    def pub(self, twist = None):
+
+    def pub(self, twist=None):
         if twist is None:
             twist = self._twist
         self._twist_pub.publish(twist)
 
-    def wait_for_robot_twisted(self, in_robot_name, in_robot_prev_pose, in_timeout=5.0):
-        assert self._is_pub_finished, f'Twist has not been published to {self.get_full_topic_name()}'
-        
+    def wait_for_robot_twisted(
+        self,
+        in_robot_name,
+        in_robot_prev_pose,
+        in_timeout=5.0,
+        publish_twist_on_fail=True,
+    ):
+        assert (
+            self._is_pub_finished
+        ), f"Twist has not been published to {self.get_full_topic_name()}"
+
         start_time = time.time()
-        while ((time.time() - start_time) < in_timeout):
+        while (time.time() - start_time) < in_timeout:
             time.sleep(1.0)
-            is_robot_found, robot_pose = wait_for_spawned_entity(in_robot_name, in_timeout=5.0)
-            assert is_robot_found, f'wait_for_robot_twisted(): {in_robot_name} unavailable!'
-            if (robot_pose != in_robot_prev_pose):
+            is_robot_found, robot_pose = wait_for_spawned_entity(
+                in_robot_name, in_timeout=5.0
+            )
+            assert (
```

Comment:
Normally assert works without brace, though it's innocuous in this case, so unclear why the format adds it so it should be used with caution
https://realpython.com/python-assert-statement

Reply:
I guess formatter changes line to meet maximum line length